libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -157,37 +181,46 @@ proc closeRemote*(s: LPChannel) {.async.} =
 method closed*(s: LPChannel): bool =
   ## this emulates half-closed behavior
   ## when closed locally writing is
-  ## dissabled - see the table in the
+  ## disabled - see the table in the
   ## header of the file
   s.closedLocal
 
-method close*(s: LPChannel) {.async, gcsafe.} =
-  if s.closedLocal:
-    return
-
-  trace "closing local lpchannel", id = s.id,
-                                   initiator = s.initiator,
-                                   name = s.name,
-                                   oid = s.oid
-  # TODO: we should install a timer that on expire
-  # will make sure the channel did close by the remote
-  # so the hald-closed flow completed, if it didn't
-  # we should send a `reset` and move on.
-  await s.closeMessage()
-  s.closedLocal = true
-  if s.atEof: # already closed by remote close parent buffer imediately
-    await procCall BufferStream(s).close()
-
-  trace "lpchannel closed local", id = s.id,
-                                  initiator = s.initiator,
-                                  name = s.name,
-                                  oid = s.oid
-
-method reset*(s: LPChannel) {.base, async.} =
+method reset*(s: LPChannel) {.base, async, gcsafe.} =
   # we asyncCheck here because the other end
   # might be dead already - reset is always
   # optimistic
   asyncCheck s.resetMessage()
   await procCall BufferStream(s).close()
   s.isEof = true
   s.closedLocal = true
+
+method close*(s: LPChannel) {.async, gcsafe.} =
+  if s.closedLocal:
+    trace "channel already closed", id = s.id,
+                                    initiator = s.initiator,
+                                    name = s.name,
+                                    oid = s.oid
+    return
+
+  proc closeRemote() {.async.} =
+    try:
+      trace "closing local lpchannel", id = s.id,
+                                       initiator = s.initiator,
+                                       name = s.name,
+                                       oid = s.oid
+      await s.closeMessage().wait(2.minutes)
```

Comment:
does `closeMessage` raise? if not, comment, if it does, `close` will not be called on exception which might be odd

Reply:
it doesn't raise, I'll add a comment.

Reply:
yes - here and elsewhere really - let's make sure the rationale for handling errors is documented at the call site consistently (either explaining what's going on or pointing to the place that does - doesn't have to be long or extensive) so that readers of the code can understand what's going on and what they should do with their own code going forwards - otherwise it's impossible to build a mental model of how the error handling is supposed to work, when reading the code - the compiler for now cannot help us in this endeavour so comments is the best we have

Reply:
yeah, thats a good idea, I'll do that.